src/service-override/storage.ts

```diff
@@ -0,0 +1,144 @@
+import { Emitter, Event } from 'vs/base/common/event'
+import { IStorage, IStorageDatabase, IStorageItemsChangeEvent, IUpdateRequest, Storage } from 'vs/base/parts/storage/common/storage'
+import { IEditorOverrideServices } from 'vs/editor/standalone/browser/standaloneServices'
+import { SyncDescriptor } from 'vs/platform/instantiation/common/descriptors'
+import { AbstractStorageService, IStorageService, StorageScope as VSStorageScope } from 'vs/platform/storage/common/storage'
+import { IUserDataProfile } from 'vs/platform/userDataProfile/common/userDataProfile'
+import { IAnyWorkspaceIdentifier } from 'vs/platform/workspace/common/workspace'
+
+export enum StorageScope {
```

Comment:
Original `StorageScope` is a `const enum` so can not be properly reexported unless we remove `const` in patch, figured that's more convenient.

Reply:
Yeah, vscode uses a lot of const enum. the build process here preserves them but the ones in monaco-editor are not preserved. I don't have any perfect solution but I've already had the same issue for another enum and I've overriden it in the `override` folder (see https://github.com/CodinGame/monaco-vscode-api/blob/8c0e697dfc8c599dc7b3499adda6122b19050f5c/src/override/vs/platform/dialogs/common/dialogs.ts)

Reply:
My reasoning to make it a string literal enum was that for implementer it removes the need to convert to strings as ultimately you would be just writing and reading these data objects and it's nice to have a string for scope already as I'd bet usually one wants a string as a name for these object keys. But yeah, if you prefer overriding like with the `ConfirmResult` - I'll change to that, just let me know

Reply:
> I understand your point, but doing it for a single enum is not very consistent. Also I think we better modify vscode code as less as possible, we don't really know what is done with that enum. There is a negative value, maybe there is a reason for it, maybe the enum value is compared somewhere, maybe it'll change in the future, who knows 🤷

Oh yes, absolutely, vscode patching must as minimal as possible. Learned that the hard way trying to maintain the fork of vscode for our needs. And that's why we're currently migrating away from vscode fork.

I will change this enum to a non-const copy of original one.